src/rainflow.py

```diff
@@ -144,7 +153,44 @@ def count_cycles(series, ndigits=None, left=False, right=False):
     counts = defaultdict(float)
     round_ = _get_round_function(ndigits)
 
-    for low, high, mult in extract_cycles(series, left=left, right=right):
-        delta = round_(abs(high - low))
-        counts[delta] += mult
+    try:
+        max_range = max(series) - min(series)
+        if max_range == 0:
+            return []
+    except ValueError:
+        return []
+
+    # check for mutually exclusive options: ndigits, nbins, binsize
+    if (ndigits is not None) and ((nbins is not None) or (binsize is not None)) :
```

Comment:
Let's squash these two if statements into one, since they both address the same issue.
```suggestion
    if sum((arg is not None) for arg in (ndigits, nbins, binsize)) > 1:
```

Reply:
Very neat :)